lib/src/main/java/edu/stanford/futuredata/macrobase/analysis/summary/IncrementalSummarizer.java

```diff
@@ -0,0 +1,320 @@
+package edu.stanford.futuredata.macrobase.analysis.summary;
+
+import edu.stanford.futuredata.macrobase.analysis.summary.count.ExactCount;
+import edu.stanford.futuredata.macrobase.analysis.summary.itemset.AttributeEncoder;
+import edu.stanford.futuredata.macrobase.analysis.summary.itemset.FPGrowth;
+import edu.stanford.futuredata.macrobase.analysis.summary.itemset.RiskRatio;
+import edu.stanford.futuredata.macrobase.analysis.summary.itemset.result.AttributeSet;
+import edu.stanford.futuredata.macrobase.analysis.summary.itemset.result.ItemsetResult;
+import edu.stanford.futuredata.macrobase.analysis.summary.itemset.result.ItemsetWithCount;
+import edu.stanford.futuredata.macrobase.datamodel.DataFrame;
+import edu.stanford.futuredata.macrobase.datamodel.Schema;
+import edu.stanford.futuredata.macrobase.operator.IncrementalOperator;
+
+import java.util.*;
+import java.util.function.DoublePredicate;
+
+/**
+ * Given a batch of rows with an outlier class column, explain the outliers using
+ * string attribute columns. Each batch is considered as an independent unit.
+ */
+public class IncrementalSummarizer implements IncrementalOperator<Explanation> {
+    // pane size known in advance
+    public int numPanes;
+
+    // Parameters
+    protected String outlierColumn = "_OUTLIER";
+    protected double minOutlierSupport = 0.1;
+    protected double minRiskRatio = 3;
+    protected boolean useAttributeCombinations = true;
+    protected List<String> attributes = new ArrayList<>();
+    protected DoublePredicate predicate = d -> d != 0.0;
+
+    // Encoder
+    protected AttributeEncoder encoder = new AttributeEncoder();
+    protected List<Set<Integer>> inlierItemsets, outlierItemsets;
+
+    // Internal booking keep
+    private Deque<HashMap<Set<Integer>, Double>> inlierItemsetPaneCounts;
+    private Deque<HashMap<Set<Integer>, Double>> outlierItemsetPaneCounts;
+    private HashMap<Set<Integer>, Double> inlierItemsetPaneCount = new HashMap<>();
+    private HashMap<Set<Integer>, Double> outlierItemsetPaneCount = new HashMap<>();
+    private HashMap<Set<Integer>, Double> inlierItemsetWindowCount = new HashMap<>();
+    private HashMap<Set<Integer>, Double> outlierItemsetWindowCount = new HashMap<>();
+    private Deque<Integer> inlierPaneCounts;
+    private Deque<Integer> outlierPaneCounts;
+    private List<Integer> inlierCountCumSum;
+    private List<Integer> outlierCountCumSum;
+    private HashMap<Set<Integer>, Integer> trackingMap = new HashMap<>();
+
+    public IncrementalSummarizer(int numPanes) { setWindowSize(numPanes); }
+
+    @Override
+    public void setWindowSize(int numPanes) {
```

Comment:
why not at construction time? what is the intended scenario where window size changes?